activerecord/lib/active_record/associations/has_many_through_association.rb

```diff
@@ -38,10 +38,12 @@ def concat_records(records)
       def insert_record(record, validate = true, raise = false)
         ensure_not_nested
 
-        if raise
-          record.save!(:validate => validate)
-        else
-          return unless record.save(:validate => validate)
+        if !record.readonly?
```

Comment:
Ah, that is true. 

After digging a bit, I found a method `#changed_for_autosave?` inside `AutosaveAssociation`. In that module, I also found this conditional:
```rb
if record.new_record? || (autosave && record.changed_for_autosave?)
```
which looks a lot like what I’m trying to accomplish here.

It makes me think: does the case of conditionally autosaving (the bug fixed by d849f42b4ecf687ed5350f5a2402fb795aa33aac) actually belong inside `AutosaveAssociation`?